lib/smart_answer/calculators/self_assessment_penalties.rb

```diff
@@ -139,14 +139,18 @@ def late_filing_penalty
     def interest
       return 0 if overdue_payment_days <= 0
 
-      days_with_penalty_interest = payment_deadline..(payment_date - 2)
+      days_with_penalty_interest = payment_deadline..(payment_date_for_tax_year)
```

Comment:
It took me quite a while to understand what was happening here with the -1 and -2.  Maybe we could define a new constant for `interest_accrual_start_date` (like we have `payment_deadline`) and use that instead of doing the -1 and -2?

Reply:
Also, thought of a very slight improvement that could be made here: there's a `prev_day` method on `Date` which might be clearer than subtracting 1?

Reply:
Yeah I didn't do a great job of making this code readable so thanks for highlighting it!

I can't use a constant as the variable we need (`payment_date`) is not available at the class level. I've restructured the method to be a bit more readable including your much better naming. I've gone with `day`/`days` terminology instead of `prev_day` due to the fact we need to state the day before yesterday too. Unless there is some magic `day_before_yesterday` method too.

Let me know if that's any clearer?